### PR TITLE
GCP: Add queue deployments

### PR DESCRIPTION
**Pull request by tjholm**

### CODE REVIEW

File: packages/plugins/gcp/src/resources/queue.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+			{
+				name: queue.name,
+			},
+			defaultResourceOptions,
+		);
+
+		new gcp.pubsub.Subscription(`${queue.name}-sub`, {
+			// XXX: Currently required relationship with pubsub queue plugin
+			name: `${queue.name}-nitricqueue`,
+			topic: this.pubsub.id,
```
Comment (davemooreuws) [inline packages/plugins/gcp/src/resources/queue.ts]: Is `this.pubsub.id` supposed to be `this.pubsub.name`? 
Ref: https://www.pulumi.com/docs/reference/pkg/gcp/pubsub/subscription/#pubsub-subscription-push

File: packages/plugins/gcp/src/resources/queue.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		// Deploy the func
+		this.pubsub = new gcp.pubsub.Topic(
+			queue.name,
+			{
+				name: queue.name,
+			},
+			defaultResourceOptions,
+		);
+
+		new gcp.pubsub.Subscription(`${queue.name}-sub`, {
```
Comment (davemooreuws) [inline packages/plugins/gcp/src/resources/queue.ts:45]: Not 100% sure about the resource name here. Might need to be changed for gcloud validation

References:
- https://cloud.google.com/pubsub/docs/admin#resource_names
- See merged PR for topic name https://github.com/nitrictech/cli/pull/215

**Review by davemooreuws — CHANGES_REQUESTED** [review]
Looks good, but I have a few comments around gcloud resource name validation and the topic reference. See review.

File: packages/plugins/gcp/src/resources/queue.ts (hunk 1 above)
Comment (tjholm) [reply packages/plugins/gcp/src/resources/queue.ts]: result is the same in both cases, I suspect id === name.

Example using id:
image

can update to name to match docs though.

File: packages/plugins/gcp/src/resources/queue.ts (hunk 2 above)
Comment (tjholm) [reply packages/plugins/gcp/src/resources/queue.ts:45]: Subscriptions can only be created under a topic resource so I'm not sure the same change in the linked PR applies. (it needs to fully qualify the resource name within the project to properly target it as an unrelated resource).

Have you had issues with a gcp deployed project with these queue changes? As far as I can tell this is all working now.

**Review by davemooreuws — APPROVED** [review]

**Review by medgar-nitric — APPROVED** [review]

**Pull request merged**